storybookjs/storybook
# Addon-docs: Fix Table of Contents heading leak

Closes #23541

## What I did

Exclude all content from Story blocks from being indexed by the Table of Contents (TOC) component by default. The user can still override the `ignoreSelector` to adjust this behaviour.

> IMO, the default headingSelector should be `h2, h3` instead of just `h3`, but this is not in the scope of this fix. Worth considering?

## How to test

1. Set up the following TOC config in the `sandbox/.storybook/preview.ts` file:
    ```ts
    import type { Preview } from '@storybook/react';
    
    const preview: Preview = {
      parameters: {
        ...
        docs: {
          toc: {
            disable: false,
            headingSelector: 'h1, h2, h3',
          },
        },
      },
    };
    
    export default preview;
    ```
2. Run the default sandbox, `yarn start`
3. Open Storybook in your browser
4. Go to the Header `Docs` page
5. See that the `Acme` headings from the sample component do not bleed into the generated Table of Contents

To test the previous behaviour, do the same thing and note the difference in Table of Contents.

## Checklist



      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md) (N/A)

#### Maintainers


`["cleanup", "BREAKING CHANGE", "feature request", "bug", "build", "documentation", "maintenance", "dependencies", "other"]`



### 🦋 Canary release



This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_
